internal/query/gin_map_test.go

```diff
@@ -10,103 +10,189 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestContextQueryNestedMap(t *testing.T) {
+func TestContextShouldGetQueryNestedMapSuccessfulParsing(t *testing.T) {
 	var emptyQueryMap map[string]interface{}
 
 	tests := map[string]struct {
 		url            string
 		expectedResult map[string]interface{}
-		exists         bool
 	}{
-		"no searched map key in query string": {
-			url:            "?foo=bar",
+		"no query params": {
+			url:            "",
 			expectedResult: emptyQueryMap,
-			exists:         false,
 		},
-		"searched map key is not a map": {
-			url:            "?mapkey=value",
-			expectedResult: emptyQueryMap,
-			exists:         false,
+		"single query param": {
+			url: "?foo=bar",
+			expectedResult: map[string]interface{}{
+				"foo": "bar",
+			},
 		},
-		"searched map key is array": {
-			url:            "?mapkey[]=value1&mapkey[]=value2",
-			expectedResult: emptyQueryMap,
-			exists:         false,
+		"multiple query param": {
+			url: "?foo=bar&mapkey=value1",
+			expectedResult: map[string]interface{}{
+				"foo":    "bar",
+				"mapkey": "value1",
+			},
 		},
-		"searched map key with invalid map access": {
-			url:            "?mapkey[key]nested=value",
-			expectedResult: emptyQueryMap,
-			exists:         false,
+		"map query param": {
+			url: "?mapkey[key]=value",
+			expectedResult: map[string]interface{}{
+				"mapkey": map[string]interface{}{
+					"key": "value",
+				},
+			},
 		},
-		"searched map key with valid and invalid map access": {
-			url: "?mapkey[key]invalidNested=value&mapkey[key][nested]=value1",
+		"nested map query param": {
+			url: "?mapkey[key][nested][moreNested]=value",
```

Comment:
Is there a maximum limit of 'Deep Nesting Limits"?
Should we consider (or secure) the extremely deep nesting `(e.g., mapkey[key][nested1][nested2][nested3]...)`?  
...to ensure there is no stack overflow or performance degradation.

Reply:
I'm setting it arbitrary to 100 `[nestings]` although on my machine it handles 100 000 pretty fast and when trying 1 000 000 it took too long for me to prepare the query string even before test 😄